### ISSUE TITLE
Postgres query with a large amount of entities fails because of regexp

### ISSUE THREAD

**Issue by ronkatz96** [typeorm/typeorm#7111]
## Issue Description
Can't execute a query with a lot of entities (checked with ~120,000 entities) using Postgres.

### Expected Behavior
The query succeeds and the entities are retuned.

### Actual Behavior
The query fails and throws the following error:

```
Invalid regular expression: ... : Too many captures
at new Regexp(<anonymous>)
at PostgresDriver.escapeQueryWithParameters(.../PostgresDriver.ts(566:27))
```

### My Environment
| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |     Windows 10     |
| Node.js version     | v13.9.0 |
| Typescript version  | v3.9.5 |
| TypeORM version     | v0.2.22 |

### Relevant Database Driver(s)
- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [ ] `mongodb`
- [ ] `mysql`
- [ ] `nativescript`
- [ ] `oracle`
- [X] `postgres`
- [ ] `react-native`
- [ ] `sap`
- [ ] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [ ] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?
- [ ] Yes, I have the time, and I know how to start.
- [ ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [X] No, I don't have the time and I wouldn't even know how to start.

**Comment by nebkat** [comment]
Have you considered performing the insert as part of multiple queries in blocks of e.g. 1000 entities, perhaps inside of a transaction? If the RegExp didn't fail you may well have encountered other issues, so such large inserts are probably best avoided.

**Comment by ronkatz96** [comment]
Thanks for the quick reply.
I'm trying to fetch the entities, not insert them.

**Comment by SAFAL12358** [comment]
Any updates on this issue?
Did you find a way around @ronkatz96 ?

**Comment by t2y** [comment]
I use version: 0.2.31 and had the same issue with hundreds of parameters.

```
SyntaxError: Invalid regular expression: /(:(\\\\.\\\\.\\\\.)?slotId0\\\\b)|(:(\\\\.\\\\.\\\\.)?slotId1\\\\b)|... (Too many captures)
at String.replace (<anonymous>),
at PostgresDriver.escapeQueryWithParameters (src/driver/postgres/PostgresDriver.ts:617:19),
```

**Comment by imnotjames** [comment]
Should be fixed by #7821
